pkg/resource/deploy/builtins.go

```diff
@@ -278,7 +278,12 @@ func (p *builtinProvider) SignalCancellation(context.Context) error {
 
 func (p *builtinProvider) readStackReference(inputs resource.PropertyMap) (resource.PropertyMap, error) {
 	name, ok := inputs["name"]
-	contract.Assertf(ok, "missing required property 'name'")
+	// If the name is not provided, we should return an error. This is probably due to a user trying to import
+	// this stack reference.
+	if !ok {
+		return nil, errors.New("stack reference can not be imported")
```

Comment:
Hrm, readStackReference is not only used when importing.  If we should never get here without the name property, shouldn't we check for that in whatever calls this for import?

Reply:
In `read`? We could, although now you mention it that could also trigger if someone messed with state. And this error could be hit be a bad invoke/create as well. All of those probably _should_ be errors rather than panics, but they probably shouldn't say anything about it being imported. I can move this check around to the top level methods instead and return appropriate errors.